deploy/2-configure-feeds.ts

```diff
@@ -17,130 +15,145 @@ interface Config {
-      pythId: pythID[networkName]["BNX"],
-      asset: assets[networkName]["BNX"],
-      maxStalePeriod: DEFAULT_STALE_PERIOD,
-    });
-    await tx.wait(1);
-
-    console.log("Configuring Resilient Oracle");
-    tx = await resilientOracle.setTokenConfig({
-      asset: assets[networkName]["BNX"],
-      oracles: [chainlinkOracle.address, pythOracle.address, addr0000],
-      enableFlagsForOracles: [true, true, false],
-    });
-  } else {
-    const mockBNX = await hre.ethers.getContract("MockBNX");
-    tx = await resilientOracle.setTokenConfig({
-      asset: mockBNX.address,
-      oracles: [chainlinkOracle.address, addr0000, pythOracle.address],
-      enableFlagsForOracles: [true, false, true],
-    });
-    await tx.wait(1);
-
-    console.log("Configuring mock Chainlink price for BNX");
-
-    tx = await resilientOracle.setOracle(mockBNX.address, chainlinkOracle.address, 0);
-    await tx.wait(1);
-
-    tx = await chainlinkOracle.setPrice(mockBNX.address, "159990000000000000000");
-    await tx.wait(1);
-
-    console.log("Configuring mock Pyth price for BNX");
-    tx = await pythOracle.setPrice(mockBNX.address, "159990000000000000000");
-    await tx.wait(1);
-  }
 
-  console.log("Configuring Bound Validator");
-  tx = await boundValidator.setValidateConfig({
-    asset: assets[networkName]["BNX"],
-    upperBoundRatio: BigNumber.from(105).pow(18),
-    lowerBoundRatio: BigNumber.from(95).pow(18),
-  });
-
-  await tx.wait(1);
-
-  //configure BSW
-  console.log("Configure BSW");
-
-  if (network.live) {
-    console.log("Configuring BSW on Pyth");
-    tx = await pythOracle.setTokenConfig({
-      pythId: pythID[networkName]["BSW"],
-      asset: assets[networkName]["BSW"],
-      maxStalePeriod: DEFAULT_STALE_PERIOD,
-    });
-    await tx.wait(1);
-  } else {
-    const mockBSW = await hre.ethers.getContract("MockBSW");
-
-    tx = await resilientOracle.setTokenConfig({
-      asset: mockBSW.address,
-      oracles: [chainlinkOracle.address, addr0000, pythOracle.address],
-      enableFlagsForOracles: [true, false, true],
-    });
-    await tx.wait(1);
-
-    console.log("Configuring mock Pyth price for BSW");
-    tx = await resilientOracle.setOracle(mockBSW.address, pythOracle.address, 0);
-    await tx.wait(1);
-
-    tx = await pythOracle.setPrice(mockBSW.address, "208000000000000000");
-    await tx.wait(1);
+  for (let i = 0; i < assets[networkName].length; i++) {
+    const asset = assets[networkName][i];
+    console.log(`Configuring ${asset.token}`);
+
+    let oracles = [];
+    let enableFlagsForOracles = [];
+
+    if (asset.oracle == "chainlink") {
+      oracles = [chainlinkOracle.address, addr0000, addr0000];
+      enableFlagsForOracles = [true, false, false];
+    } else {
+      oracles = [pythOracle.address, addr0000, addr0000];
+      enableFlagsForOracles = [true, false, false];
+    }
+
+    if (network.live) {
+      let tx;
+
+      if (asset.oracle == "chainlink") {
```

Comment:
An easy pattern that could increase the maintainability of the code could be to extract the data associated with the oracles to a map, using `asset.oracle` as the key. For example, we could define a map like this one:

```
const oraclesData = {
	chainlink: {
		oracles: [chainlinkOracle.address, addr0000, addr0000],
		enableFlagsForOracles: [true, false, false],
		underlyingOracle: chainlinkOracle,
		getTokenConfig: (asset, networkName) => ({
			asset: asset.address,
          	feed: chainlinkFeed[networkName][asset.token],
          	maxStalePeriod: DEFAULT_STALE_PERIOD,
		}),
	},
	pyth: {
		oracles: [pythOracle.address, addr0000, addr0000],
		enableFlagsForOracles: [true, false, false],
		underlyingOracle: pythOracle,
		getTokenConfig: (asset, networkName) => ({
			pythId: pythID[networkName][asset.token],
          	asset: asset.address,
         	maxStalePeriod: DEFAULT_STALE_PERIOD,
		}),
	}
}; 
```

And then remove the if/else statements, replacing them by direct calls using the map, like:
```
const oracle = oraclesData[asset.oracle];
tx = await oracle.underlyingOracle.setTokenConfig(oracle.getTokenConfig(asset, networkName));
```

Up to you, it is just a suggestion...

Reply:
I think we cannot use a map here because we need to wait for each transaction to be mined before sending a new transaction. `await` inside map won't work.  If I make the map return a array of promises then all txns will have same `nonce` and likely to fail.

Reply:
I have refactored the code based on your suggestion.